pkg/logql/log/distinct.go

```diff
@@ -0,0 +1,37 @@
+package log
+
+type distinctFilter struct {
+	datas  map[string]map[string]int
+	labels []string
+}
+
+func NewDistinctFilter(labels []string) (Stage, error) {
+	datas := make(map[string]map[string]int, 0)
+	for _, label := range labels {
+		datas[label] = make(map[string]int, 0)
+	}
+	return &distinctFilter{
+		labels: labels,
+		datas:  datas,
+	}, nil
+}
+func (r *distinctFilter) Process(_ int64, line []byte, lbs *LabelsBuilder) ([]byte, bool) {
+	isPass := false
+	for _, label := range r.labels {
+		val, ok := lbs.Get(label)
+		if !ok {
+			return line, false
+		}
+		_, ok = r.datas[label][val]
+		if ok {
+			return line, false
```

Comment:
if `r.datas[label][val]` is 1 why don't you have pass=true here?

ex:
1. starting with lbs={"a": "bla"}, r.datas={"a": "bla"}
2.  lbs.Get("a") will return `"bla", true` for line 21
3. r.datas["a"]["bla"] will have `ok=true` for line 25
4. instead of dropping the log line, because you had already seen that label value, you actually return with pass=false

Reply:
pass is "pass this line forwards"

It looks like this var name is misunderstood, can you help me with a better var name

Reply:
I think "keep" is a little more clear

Reply:
done, thanks  😃